### DISCUSSION TITLE
How do you structure your notes?

### DISCUSSION THREAD

**Discussion by riccardoferretti** [foambubble/foam#1081] [General]
Many times users ask for advice on how to structure their notes, so I thought it'd be interesting to have a place to point them at.

**Comment by riccardoferretti** [comment]
I think a good starting point can be the very [foam-template](https://github.com/foambubble/foam-template), both in the directory structure and in the small guide it provides to get people started with note taking.

My current approach is more similar to [PARA](https://fortelabs.com/blog/para/), with various changes that have accumulated over time.

I know of other people using [Zettelkasten](https://zettelkasten.de/posts/overview/), or GTD, Bullet Journal, ... and I am sure there are many who use none or their own.

A few high level thoughts on (my) note taking method:
1. although I do have an approach, it has evolved over time, and I don't consider it perfect, and I know it will change again in the future
2. the way I evolve my note taking is simply by taking notes and using them... see where things fall through the cracks and fix the cracks :) (a crack can be me not being able to find some content that I know is there somewhere, or not knowing where I should really put a note, ...)
3. sometimes I find the solution to the crack by tweaking the way i structure notes, and sometimes by tweaking the tools that I use (including Foam)

**Reply by xdaluca** [reply]
Interesting thoughts, the template is indeed a good start, but it would be quite helpful to create some foam templates for people that use the Zettelkasten method, Para and list examples of Foam second brains. 
Any thoughts on collaborative digital gardens/second brains?

**Reply by riccardoferretti** [reply]
Nothing specific on my side on collaborative digital gardens (except keeping them somewhere like git/dropbox), maybe others have had such use case?

**Reply by theAkito** [reply]
Greetings 👋🏻

Thank you for the P.A.R.A. method. It looks very intriguing! 🙂

> My current approach is more similar to [PARA](https://fortelabs.com/blog/para/), with various changes that have accumulated over time.

I would like to add the following explanation for that method.

https://www.lucapallotta.com/para/

I think, this explains it much better. I had lots of open questions, when I read the original post you linked above.
Then, I read the second post written by Luca and all my questions were answered.

**Comment by ghost** [comment]
Hello, (for context I've been using foam as my main note-taking tool for maybe a year, and am happy with it and seems normal to use, i.e. I'm not struggling against something, and it has helped me organize and retrieve notes/knowledge, hence why i'm sharing it now. but its nothing revolutionary) so I've tried several structured methods like Zettelkasten, GTD, PARA, etc. But in the end, i often go back to something that seems more "natural/organic" to me, at least for now. My directories are actually divided in projects/areas/journal/archive (so kinda based on PARA but organically evolved out). But **it's mainly divided in two groups: permanent notes (my areas) and my more temporary projects (projects/journal)**

1. For my permanent notes (my areas folder) i have a list of subjects that I'm interested in, one for each file. I retrieve them with ctrl+P and type the name. It can be whatever, "data science", "politics", "psychology", "cooking", "habits" (I've tried to go more granular in notes but then i kinda forget they exist, like a specific economic theory, so now i'll just add it in a part of the economy note). In these I kinda structure notes in chapters and sub-chapters (with markdown headings, so i use vscode's outline sidebar a lot), then either paragraphs with lots of linebreaks or in lists/sublists. When i'll come across an interesting idea/thought, etc., I'll add it on the note in its subpart with a ref/link when applicable (I can then reorganize parts if they become too big). **So basically for permanent notes : list of big notes by subject that i find with ctrl+P**.
2. For my projects/temporary notes, I have a file that was initially based on some GTD method but is now just a checklist/sublist which is kinda classified by project. In it I'll list the tasks and projects i'm doing, and then link to some permanent note (for a longer project), or to a daily note that i'll have renamed. For instance if I have a task of writing a report, I could create a daily note with alt+D, to which name i'll append some description, so it'll become something like: "2022-10-13_report_companyX.md", and it will be linked in the todo file/checklist. Foam works well with that in recreating correct links when i rename notes. **So basically for projects: a "hub" file which links and unlinks to temporary files**.

This totals to ~140 notes right now, with about ~90 permanent notes and the rest for project/daily.

To input notes (my "inbox") if I'm not on the computer i often actually just note it somewhere else, then add it to foam later, else i just go to foam, ctrl+p the relevant note (or the todo file), and add the thing.

What i'm lacking for now in my ever evolving workflow: sometimes I maybe want to input things different than just a linear text. If I'm brainstorming or noting on paper, i can easily underline, add stuff side by side, in margins, in footnotes, draw arrows, etc. maybe i could try to use markdown tables more, or color stuff with highlighting extensions. But so I still write stuff in physical notepads for specific bigger tasks that i then recap in foam. Also sometimes i like writing bigger reports in libreoffice writer or some latex file, with my foam system on the side of the screen for notes, but i maybe wont write it directly in foam. not sure why, maybe Writer seems more "writing long reports"-like.. depends on the times.
Also, in the coming times I'll have to organize some scientific literature readings and note-taking, so it'll likely evolve, maybe with more footnotes, zotero reference linkings and more folders. As well as more hubs maybe.

have a nice day

**Comment by chrisUsick** [comment]
I’m happy that @riccardoferretti started this conversation. 
I've been taking notes in various apps for 3ish years now. Currently, school of note taking is a second brain approach. 
More than focus on a specific methodology, I think it’s important to also focus on learning “why” you capture notes. 
Tiago Forte has been the primary source of education for me. 

Now, back to the question of how I take notes:
I’ve taken a radical “network thinking” approach that focuses on ease of capturing and an evolutionary organization approach. 
Instead of making any folders I exclusively use notes and links. 
I have high level “hub” notes, that generally align to a broad topic, ex “software development” or a project. 
Then, whether I’m capturing an idea, some research, a useful article, I simply create a new note, and link it to any other notes I can think of as relevant. 
I have found this to a very effective way to combat the never ending issue of deciding where to “put” a note.

**Comment by clux** [comment]
Hey all. Wanted to say thank you to the maintainers of foam for making a free and completely open ecosystem!

I used foam heavily for managing a D&D campaign ([repo link](https://github.com/clux/facemaulers)) between friends, and even in my simple use i found it very powerful.
Used [folders for basic categorisations](https://github.com/clux/facemaulers/tree/main/docs), no tags - just a lot of cross linking - [hijacked the notes](https://github.com/clux/facemaulers/blob/main/.foam/templates/recap-template.md) for recaps, [simple shell scripting](https://github.com/clux/facemaulers/blob/main/stats.sh) for dataview-like reductions, and at the end of the campaign published it all online by [hooking the brain up](https://github.com/clux/facemaulers/blob/main/mkdocs.yml) to [mkdocs-material](https://squidfunk.github.io/mkdocs-material/) which [ended up looking pretty professional](https://clux.github.io/facemaulers/) imo. Wrote a [blog post about my experience](https://clux.dev/post/2021-12-05-campaign-concluded/) at the time.

Am also using it as a note taking helper for my personal second brain experiment, but because it is a light-weight tool I am leaning more on external solutions for stuff not explicitly related to note-taking (e.g. private github kanban boards for task/progress tracking).
